tests/functional/raw-node/test/routes/unsupportedQuries.js

```diff
@@ -0,0 +1,34 @@
+import assert from 'assert';
+import constants from '../../../../../constants';
+import { makeS3Request } from '../../utils/makeRequest';
+
+const bucket = 'testunsupportedqueriesbucket';
+const objectKey = 'key';
+
+const itSkipIfAWS = process.env.AWS_ON_AIR ? it.skip : it;
+
+describe('unsupported query requests:', () => {
+    constants.unsupportedQueries.forEach(query => {
+        itSkipIfAWS(`should respond with NotImplemented for ?${query} request`,
+        done => {
+            const queryObj = {};
+            queryObj[query] = '';
+            makeS3Request({ method: 'GET', queryObj, bucket, objectKey },
+            err => {
+                assert.strictEqual(err.code, 'NotImplemented');
+                assert.strictEqual(err.statusCode, 501);
+                done();
+            });
+        });
+    });
+
+    itSkipIfAWS('should accept blacklisted query field as a value ' +
```

Comment:
should accept blacklisted query key as a query value to a query key that is not on the blacklist